code/game/objects/items/devices/scanners.dm

```diff
@@ -426,12 +426,12 @@ GENE SCANNER
 				message += "<span class='notice'>[round(R.volume, 0.001)] units of [R.name][R.overdosed == 1 ? "</span> - <span class='boldannounce'>OVERDOSING</span>" : ".</span>"]"
 		else
 			message += "<span class='notice'>Subject contains no reagents.</span>"
-		if(M.reagents.addiction_list.len)
-			message += "<span class='boldannounce'>Subject is addicted to the following reagents:</span>"
-			for(var/datum/reagent/R in M.reagents.addiction_list)
-				message += "<span class='alert'>[R.name]</span>"
+		if(LAZYLEN(M.mind.active_addictions))
```

Comment:
LAZYLEN? why not just use `length()`

Reply:
Because active addictions is a lazy list. There is no behavioural difference, but LAZYLEN is used to indicate that something being accessed is a lazylist.
Any non-lazy list using LAZYLEN is incorrect and any lazy list using length() is incorrect.